### PR TITLE
Refactor to use `node-ipc` protocol in tests

### PR DESCRIPTION
**Pull request by remcohaszing**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

This allows us to use the console, which is useful for debugging.

This also simplifies how paths are resolved in tests. I combined these changes, because they touch the same lines.

### CODE REVIEW

**Review by wooorm — COMMENTED** [review]
- Looks like this fails on windows
- Can you clarify how the console can be used?
- Otherwise fine

File: test/index.js [hunk 1]
```diff
@@ -812,21 +813,19 @@ function cleanStack(stack, max) {
 function startLanguageServer(t, serverFilePath, cwd = '.') {
```
Comment (wooorm) [inline test/index.js:813]: The code that calls `startLanguageServer` seems to pass `cwd` as a file path: E.g., https://github.com/remcohaszing/unified-language-server/blob/tests-node-ipc/test/index.js#L567.
But here it is treated as a URL?

Comment (remcohaszing) [reply test/index.js]: I think you’re right, but I decided to revert this part. I’ll look into it in another PR.

**Comment by remcohaszing** [comment]
> Can you clarify how the console can be used?

`stdout` and `stderr` are now inherited from the parent process. It’s not used by the language client anymore. This means you can just write anything using `console.log()`, `console.dir()`, or whatever you like in the test and it will show up in the terminal.

**Review by wooorm — APPROVED** [review]

**Pull request merged**